diff --git a/binding.gyp b/binding.gyp
index 81a1238c4ed85f2049c850bf2d3f790f8fb8c6e9..f81970bca9ef5f8ffe57c23424958fdd8b214fc1 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -13,6 +13,11 @@
       'xcode_settings': {
         'OTHER_CPLUSPLUSFLAGS': ['-std=c++20', '-stdlib=libc++'],
       },
+      'configurations': {
+        'Release': {
+          'msbuild_toolset': 'v143',
+        },
+      },
       'msvs_settings': {
         'VCCLCompilerTool': {
           'AdditionalOptions': [
@@ -33,6 +38,11 @@
       'target_name': 'test_extension',
       'dependencies': ['deps/sqlite3.gyp:sqlite3'],
       'conditions': [['sqlite3 == ""', { 'sources': ['deps/test_extension.c'] }]],
+      'configurations': {
+        'Release': {
+          'msbuild_toolset': 'v143',
+        },
+      },
     },
   ],
 }
diff --git a/deps/common.gypi b/deps/common.gypi
index 613adc358e9b2577366453409ee3ef145cf5d166..5c749e09a2c489e7389066e105338be0b36c60ea 100644
--- a/deps/common.gypi
+++ b/deps/common.gypi
@@ -62,6 +62,12 @@
           'DEAD_CODE_STRIPPING': 'YES',
           'GCC_INLINES_ARE_PRIVATE_EXTERN': 'YES',
         },
+        'msbuild_toolset': 'v143',
+        'msvs_settings': {
+          'VCLinkerTool': {
+            'AdditionalOptions': ['/ignore:4199'],
+          },
+        },
       },
     },
   },
diff --git a/deps/sqlite3.gyp b/deps/sqlite3.gyp
index 7259dfef74d0cec4d4d4bf225ab5acd78436cca7..2c0f04fca2b84a02f1f1d2d427d9c6012c122014 100644
--- a/deps/sqlite3.gyp
+++ b/deps/sqlite3.gyp
@@ -72,6 +72,7 @@
           'msvs_settings': { 'VCCLCompilerTool': { 'RuntimeLibrary': 1 } }, # static debug
         },
         'Release': {
+          'msbuild_toolset': 'v143',
           'msvs_settings': { 'VCCLCompilerTool': { 'RuntimeLibrary': 0 } }, # static release
         },
       },
